doc/adr/0005-tsp-update-experiment.md

```diff
@@ -0,0 +1,58 @@
+# 5. Tsp update experiment
+
+Date: 2022-03-22
+
+## Status
+
+Proposed
+
+## Context
+
+The tentative [TSP specification][tsp] proposes a PUT command on the `tsp/experiments/{expUUID}` endpoint for the following use cases:
+
+- Rename experiment
+- Add/remove traces to/from an experiment
+- Other updates (e.g. offsetting of traces)
+
+When an experiment is created the client passes the `name` of the experiment and a `list of trace UUIDs`  to the POST request.
+It returns a data structure with experiment details and an `unique UUID`, to identify the experiment.
+
+This ADR proposes the way-forward for renaming and adding/removing of traces of experiments. It doesn't cover other updates like offsetting of traces. They will be handled in other ADRs.
+
+PUT commands require to be idempotent, multiple requests have to have the same outcome. For a PUT to be idempotent, it is required to provide all parameters that makes up an experiment to be sent in the PUT command. In the case of updating of an experiment, the `name` and `list of trace UUIDs` have to be present, even if some parts don't change. The PUT command replaces the experiment resource on the server with a new one.
```

Comment:
If `name` is not present in an hypothetical PUT request, then maybe `trace-server` could simply ignore touching the name of that experiment, entirely. Such a `name`-less PUT request would remain idempotent that way. Same could go for the list of trace UUIDs; if present, touch that list consistently, and if not (present in the request), just don't.